python/cudf/cudf/core/dataframe.py

```diff
@@ -543,6 +543,9 @@ def to_string(self):
     def __str__(self):
         return self.to_string()
 
+    def astype(self, dtype, **kwargs):
```

Comment:
It would be nice to allow the `errors='ignore'` or `errors='raise'` behavior in this PR as well, though probably not enough to hold up the PR if it adds too much complexity for now.

Reply:
I added this, let me know if this is what you had in mind.

Reply:
@brandon-b-miller looks good! Do you think it's worth exploring `astype` `copy` behavior in this PR as well, or perhaps a follow on?

Reply:
good question - would be nice to have for sure, I'll pop the hood and take a look.

Reply:
@beckernick I am going to handle the copy behavior separately so as to unblock a separate PR that depends on this.

Reply:
Makes sense @brandon-b-miller . Thanks for exploring it in this PR 👍